src/Microsoft.AspNet.Http.Extensions/ResponseExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Http.Extensions
+{
+    public static class ResponseExtensions
```

Comment:
I think we should throw a custom exception if we know we're in an invalid state.